lib/components/term.js

```diff
@@ -144,43 +144,43 @@ export default class Term extends React.PureComponent {
 
     if (!props.term) {
       this.term.attachCustomKeyEventHandler(this.keyboardHandler);
+      this.term.loadAddon(fitAddon);
+      this.term.loadAddon(searchAddon);
+      this.term.loadAddon(webLinksAddon);
       this.term.open(this.termRef);
       fitAddon.fit();
+      this.term.loadAddon(webGlAddon);
       // this.term.webLinksInit();
       // this.term.winptyCompatInit();
-      this.loadAddon(fitAddon);
-      this.loadAddon(webGlAddon);
-      this.loadAddon(searchAddon);
-      this.loadAddon(webLinksAddon);
     }
 
     if (this.props.isTermActive) {
       this.term.focus();
     }
 
     if (props.onTitle) {
-      this.disposableListeners.push(this.term.addDisposableListener('title', props.onTitle));
+      this.disposableListeners.push(this.term.onTitleChange(props.onTitle));
     }
 
-    if (props.onActive) {
-      this.disposableListeners.push(this.term.addDisposableListener('focus', props.onActive));
-    }
+    // if (props.onActive) {
+    //   this.disposableListeners.push(this.term.addDisposableListener('focus', props.onActive));
```

Comment:
This might need a bit of digging. I was not able to find any suitable solution when I was looking at this.

Reply:
yah same :/

Reply:
perhaps use dom listener instead. https://github.com/microsoft/vscode/blob/c46560d371702e8d6ab063e46d254b0b8afeed54/src/vs/workbench/contrib/terminal/browser/terminalInstance.ts#L639

Reply:
Yes the focus event was removed because you can just listen to focus on `Terminal.element`, see https://github.com/xtermjs/xterm.js/pull/2008#issue-269040997

> blur -> Removed as you can just listen to the DOM event on Terminal.textarea
> focus -> Removed as you can just listen to the DOM event on Terminal.textarea